docs/configuration/providers/index.md

```diff
@@ -0,0 +1,38 @@
+---
+title: Configuration Providers
+layout: default
+---
+
+Configuration can typically come from different sources. For exemple, command line or desktop apps usually store the configuration in files the the user home directory; server deployments might use a central configuration database; or during testing, the configuration might be hardcoded in the test.
+
+To implement this behavior, Powsybl defines a Configuration Provider interface: `com.powsybl.commons.config.PlatformConfigProvider`. You can use the implementations shipped with Powsybl or implement your own.
+
+Note: The `iTools` and `GSE` distributions automatically use the `ClassicPlatformConfigProvider` so you don't have to think about it when using these programs. But when using Powsybl as a plain java library, the developper must choose a configuration provider. To choose a configuration provider, ensure that it is the only implementation of PlatformConfigProvider available to Java's ServiceLoader. Otherwise, you will get the following error:
+```
+com.powsybl.commons.PowsyblException: Platform configuration provider not found
+    at com.powsybl.commons.config.PlatformConfig.defaultConfig (PlatformConfig.java:70)
+    [...]
+```
+
+
+## List of Configuration Providers
+
+### Classic Config
+The ClassicPlatformConfigProvider is described in [Modules configuration](../modules/). It is the historical and main way to configure Powsybl. Add the following dependency in the pom.xml of your project to use it:
+```xml
+<dependency>
+    <groupId>com.powsybl</groupId>
+    <artifactId>powsybl-config-classic</artifactId>
+    <version>${powsybl.version}</version>
+</dependency>
+```
+
+### Test Config
+The TestPlatformConfigProvider reads all the config from the classpath, making the test indepedant from the user's filesystem. This is done to have reproducible tests. Add the following dependency in the pom.xml of your project to use it:
```

Comment:
Could you explain more how it works? where to put the config.yml file? How to access to other resources (using the filelist)?

Reply:
done